flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -1943,6 +1943,147 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "false")
   }
 
+  @Test
+  def testTimestampDiff(): Unit = {
+    testTableApi(
+      "2018-07-05 11:11:11".toTimestamp - "2018-07-03 11:11:11".toTimestamp,
+      "'2018-07-05 11:11:11'.toTimestamp - '2018-07-03 11:11:11'.toTimestamp",
+      "172800000"
+    )
+
+    testTableApi(
+      "2018-07-05 11:11:11".toTimestamp - "2018-07-03".toDate,
```

Comment:
Do we really want to support subtracting a date from a timestamp? Is the result useful? Shouldn't the result be another timestamp?